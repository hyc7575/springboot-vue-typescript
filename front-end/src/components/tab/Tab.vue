<template>
	<div>
		<ul>
			<li v-for="(title, index) in tabs.map(tab => tab.title)" v-bind:key="index">
				<button @click="changeCurrentTabIndex(index)" v-bind:class="[{ 'tab-title-current' : index === currentTabIndex }]">
	        {{title}}
	      </button>
			</li>
		</ul>
		<div v-for="(content, index) in tabs.map(tab => tab.content)" v-bind:key="index" v-bind:style="{ display: index === currentTabIndex ? 'block' : 'none' }">
			{{content}}
		</div>
	</div>
</template>

<script lang="ts">
	import Vue from "vue";
	import Component from "vue-class-component";

	interface TabSampleInterface {
		currentTabIndex: number;
		tabs: Array < {
			title: string;
			content: string;
		} > ;
		changeCurrentTabIndex(index: number): void;
	}

	@Component({})
	export default class TabSample extends Vue implements TabSampleInterface {
		currentTabIndex = 0;
		tabs = [{
				title: "tab1-title",
				content: "tab1-content"
			},
			{
				title: "tab2-title",
				content: "tab2-content"
			},
			{
				title: "tab3-title",
				content: "tab3-content"
			}
		];
		changeCurrentTabIndex(index: number) {
			this.currentTabIndex = index;
		}
	}
</script>

<style>
	.tab-title-current {
		background-color: orange;
	}
</style>
