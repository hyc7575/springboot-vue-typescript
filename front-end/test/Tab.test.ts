import { mount } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';
import Tab from "../src/components/tab/Tab.vue";

describe('Tab.vue', () => {
    const renderer = createRenderer();
    it('should initialize correctly', () => {
        const wrapper = mount(Tab);

        renderer.renderToString(wrapper.vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    })
});
