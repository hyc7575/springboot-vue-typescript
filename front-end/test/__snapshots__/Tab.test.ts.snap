// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab.vue should initialize correctly 1`] = `
<div>
    <ul>
        <li>
            <button class="tab-title-current">
                tab1-title
            </button>
        </li>
        <li>
            <button>
                tab2-title
            </button>
        </li>
        <li>
            <button>
                tab3-title
            </button>
        </li>
    </ul>
    <div style="display:block;">
        tab1-content
    </div>
    <div style="display:none;">
        tab2-content
    </div>
    <div style="display:none;">
        tab3-content
    </div>
</div>
`;
